src/Xamarin.Android.Build.Tasks/Tasks/WriteLockFile.cs

```diff
@@ -0,0 +1,51 @@
+using System;
+using System.IO;
+using Microsoft.Build.Framework;
+using Microsoft.Build.Utilities;
+
+namespace Xamarin.Android.Tasks
+{
+	/// <summary>
+	/// Generates a warning if LockFile exists, registers the file to be deleted at end of build
+	/// </summary>
+	public class WriteLockFile : Task
+	{
+		[Required]
+		public string LockFile { get; set; }
+
+		public override bool Execute ()
+		{
+			var path = Path.GetFullPath (LockFile);
+			var key = new Tuple<string, string> (nameof (WriteLockFile), path); // Use the full path as part of the key
+
+			// Check if already registered, for sanity
+			var existing = BuildEngine4.GetRegisteredTaskObject (key, RegisteredTaskObjectLifetime.Build) as DeleteFileAfterBuild;
+			if (existing == null) {
+				if (File.Exists (path)) {
+					Log.LogCodedWarning ("XA5302", "Two processes may be building this project at once. Lock file exists at path: {0}", path);
```

Comment:
Suggestion?  Should the lock file contain the pid of the first process to create the file?  That way the message could print which process id started the build and what the current process id is.

However, as we believe that this is largely for intra-IDE scenarios, that might not actually be incredibly useful...

Reply:
I don't think writing the pid helps us that much. If I was actively diagnosing this warning, I think I could figure out where the rogue MSBuild was coming from.